helmclient.go

```diff
@@ -116,68 +138,237 @@ func (c *Client) GetReleaseHistory(releaseName string) (*ReleaseHistory, error)
 		return nil, microerror.Maskf(tooManyResultsError, "%d releases found, expected 1", len(resp.Releases))
 	}
 
-	release := resp.Releases[0]
-	if release.Chart != nil && release.Chart.Metadata != nil {
-		version = release.Chart.Metadata.Version
-	}
+	var history *ReleaseHistory
+	{
+		release := resp.Releases[0]
 
-	history := &ReleaseHistory{
-		Name:    release.Name,
-		Version: version,
+		var version string
+		if release.Chart != nil && release.Chart.Metadata != nil {
+			version = release.Chart.Metadata.Version
+		}
+
+		history = &ReleaseHistory{
+			Name:    release.Name,
+			Version: version,
+		}
 	}
 
 	return history, nil
 }
 
 // InstallFromTarball installs a chart packaged in the given tarball.
 func (c *Client) InstallFromTarball(path, ns string, options ...helmclient.InstallOption) error {
-	_, err := c.helmClient.InstallRelease(path, ns, options...)
+	t, err := c.newTunnel()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	defer c.closeTunnel(t)
+
+	_, err = c.newHelmClientFromTunnel(t).InstallRelease(path, ns, options...)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
 	return nil
 }
 
+func (c *Client) InstallTiller() error {
+	var name = "tiller"
+	var namespace = "kube-system"
+
+	// Create the service account for tiller so it can pull images and do its do.
+	{
+		n := namespace
+		i := &corev1.ServiceAccount{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "ServiceAccount",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		}
+
+		_, err := c.k8sClient.CoreV1().ServiceAccounts(n).Create(i)
+		if errors.IsAlreadyExists(err) {
+			// fall through
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	// Create the cluster role binding for tiller so it is allowed to do its job.
+	{
+		i := &rbacv1.ClusterRoleBinding{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "rbac.authorization.k8s.io/v1",
+				Kind:       "ClusterRoleBinding",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      name,
+					Namespace: namespace,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     "cluster-admin",
+			},
+		}
+
+		_, err := c.k8sClient.RbacV1().ClusterRoleBindings().Create(i)
+		if errors.IsAlreadyExists(err) {
+			// fall through
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	// Install the tiller deployment in the guest cluster.
+	{
+		o := &installer.Options{
+			ImageSpec:      "gcr.io/kubernetes-helm/tiller:v2.8.2",
```

Comment:
Constants here for image and tag would be nice too.

Reply:
Sorted.